### ISSUE TITLE
UntrustedArgumentDiagnosticContainmentTests orphan-line rule rejects the cache-eviction notice it asks for

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3747]
`AssertEveryLineIsOwned` (`src/dotnet-inspect.Tests/UntrustedArgumentDiagnosticContainmentTests.cs:240`) says "Every stderr line must be indented or written by `CommandError`", but the predicate never checks authorship — it accepts a line only when it is indented or starts with `Error: `, `Warning: `, or `Note: `.

The cache-maintenance notice at `src/dotnet-inspect/Program.cs:293` *is* written by `CommandError.WriteLine`, and starts with `Removed `:

```
Removed 51.6 MB from obsolete cache entries.
```

So it is reported as an orphan and the test fails — but only when `cacheMaintenance.BytesFreed > 0`, which depends on shared `~/.cache/dotnet-inspect` state rather than on anything the test controls.

Observed as a nondeterministic failure of `Diagnostic_KeepsItsDetailBlockAndNeverGrowsALineFromItsMessage`: 5 failures in 8 consecutive runs while the cache was over budget, with the freed amount shrinking each time (51.6 MB, then 1.2 MB, then 475.1 KB), then 0 failures in 6 runs once it had drained. That decay is the tell — the test is reading shared machine state.

Pre-existing and unrelated to any generic-constraint work; found while validating #3727.

Either make the predicate check authorship as its message claims, or run the command under a cache root the test owns so maintenance is deterministic.
